### PR TITLE
Added QUnit tests to project

### PR DESCRIPTION
**Pull request by effrenus**
Update package.json: new build/test commands in `scripts` section; write some information about author and maintainer.

Write some tests using QUnit framework, run via karma. To run tests type in console `npm test`, will be launched Chrome instance. Possible to execute in browserstack, saucelabs. Using Chrome because phantomJS doesn't support Audio API.

Remove jPlayer `stop` event handler, because this event doesn't exist (see http://jplayer.org/latest/developer-guide/#jPlayer-events). Am I wrong?

### CODE REVIEW

**Comment by cakeinpanic** [comment]
@effrenus  Why not using mock for jplayer to use phantomjs?

**Comment by effrenus** [comment]
Bright idea, I'll try=) Maybe drop support for old browsers and use Audio API, so we could remove jplayer.

**Comment by effrenus** [comment]
@cakeinpanic done

File: test/jplayer.mock.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		this.status = $.extend({}, DEFAULT_STATUS, {duration: TRACK_DURATION_TIME * 1000});
+		this.fromTime = 0;
+
+		setTimeout($.proxy(this.onReady, this), 10);
+	}
+
+	$.jPlayer.event = {};
+
+	$.each([
+			'ready', 'setmedia', 'loadstart', 'progress', 'suspend', 'abort', 'emptied', 'stalled',
```
Comment (cakeinpanic) [inline test/jplayer.mock.js]: not so sure this large amount of literals is really needed. Removed one randomly, all test were still passing

**Comment by cakeinpanic** [comment]
I think your jplayer mock is too large for this tests. All we need are mocks for methods used in jouele code, not for all of them.
I've found these in `src/jouele.js`:
`playHead, pause, stop, play, setMedia` (maybe missed some)

File: test/jplayer.mock.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+		play: function () {
+			if (!this.mp3) {
+				this.onError({reason: 'Set media before play'});
+				return;
+			}
+
+			this.isPlaying = true;
+			this.startTime = new Date();
+			this.timerId = setInterval($.proxy(this.emulatePlay, this), UPDATE_INTERVAL * 1000);
```
Comment (cakeinpanic) [inline test/jplayer.mock.js]: Mock shouldn't contain such complex things like $.proxy, it's just a mock :)
It should  just pretend to be working, not work at all

**Comment by cakeinpanic** [comment]
Also commit message `merge` is not very informative, i'd suggest to leave default git message with information about branches being merged

**Comment by effrenus** [comment]
Thanks for review. Is there other parts that should be tested? Now coverage is around 48%

File: test/jplayer.mock.js (hunk 1 above)
Comment (effrenus) [inline test/jplayer.mock.js]: Yep, you are right, only few of them are truly needed.

File: test/jplayer.mock.js (hunk 2 above)
Comment (effrenus) [inline test/jplayer.mock.js]: Maybe its skeleton for future player) Used `$.proxy` just to bind `this` to function. 
And only for `playHead, pause, stop, play, setMedia` I've created mocks, other methods for convenience.

**Comment by cakeinpanic** [comment]
Perfect test are tests with 100% coverage(but not all test with 100% coverage are perfect :)). I suggest you to use coverageReporter type `html` and look what methods are not reached by tests and fulfill these gaps.

File: karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['qunit'],
+
+    // list of files / patterns to load in the browser
+    files: [
+      'src/jouele.css',
+      'src/jouele.skin.css',
+      'http://code.jquery.com/jquery-1.11.3.min.js',
```
Comment (cakeinpanic) [inline karma.conf.js]: bad practice, tests won't run without internet, how you're going to work during the flight?) add jquery to npm devDependencies and se local files here.

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '',
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['qunit'],
+
+    // list of files / patterns to load in the browser
+    files: [
+      'src/jouele.css',
```
Comment (cakeinpanic) [inline karma.conf.js]: Why would you need any css for phantomjs?)

File: test/emulation-check.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Comment (cakeinpanic) [inline test/emulation-check.html]: Didn't get what this file is for. To check that jplayer mock works correctly? It's not needed in repo, why should we be testing tests?)

File: test/jplayer.mock.js (hunk 2 above)
Comment (cakeinpanic) [inline test/jplayer.mock.js]: Hmm, i don't feel any convenience looking at mock with million methods including some future skeleton) 
To prove my point i'll try to create mock as i mean it and send you PR to your PR)

**Comment by cakeinpanic** [comment]
Also i've discovered that if I remove first test, all other are failing. It's not ok when any test depends on previous ones

**Pull request closed without merging**